addons/website/static/src/js/content/menu.js

```diff
@@ -660,6 +657,130 @@ publicWidget.registry.hoverableDropdown = animations.Animation.extend({
+        const hasMegaMenu = !!megaMenuToggleEl.parentElement.querySelector(".o_mega_menu");
+        if (hasMegaMenu) {
+            return;
+        }
+        this.options.wysiwyg && this.options.wysiwyg.odooEditor.observerUnactive("moveMegaMenu");
+        let megaMenuEl, previousDropdownEl;
+        const megaMenuId = megaMenuToggleEl.querySelector("span").dataset.oeId;
+        // If the `data-oe-id` attribute is there, we directly use it to find
+        // the corresponding mega menu (to be more efficient). Otherwise, we use
+        // the menus order.
+        if (megaMenuId) {
+            megaMenuEl = this.el.querySelector(`.o_mega_menu[data-oe-id="${megaMenuId}"]`);
+            previousDropdownEl = megaMenuEl.parentElement.querySelector(".o_mega_menu_toggle");
+        } else {
+            const isMobileNavbar = !!megaMenuToggleEl.closest(".navbar.o_header_mobile");
+            const currentNavbarToggleEls = isMobileNavbar ? this.mobileMegaMenuToggleEls : this.desktopMegaMenuToggleEls;
+            const otherNavbarToggleEls = isMobileNavbar ? this.desktopMegaMenuToggleEls : this.mobileMegaMenuToggleEls;
+            const megaMenuToggleIndex = currentNavbarToggleEls.indexOf(megaMenuToggleEl);
+            previousDropdownEl = otherNavbarToggleEls[megaMenuToggleIndex];
+            megaMenuEl = previousDropdownEl.parentElement.querySelector(".o_mega_menu");
+        }
+        // Hiding the previous dropdown before moving the mega menu.
+        Dropdown.getOrCreateInstance(previousDropdownEl).hide();
+        megaMenuToggleEl.insertAdjacentElement("afterend", megaMenuEl);
+        this.options.wysiwyg && this.options.wysiwyg.odooEditor.observerActive("moveMegaMenu");
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Called when a mega menu dropdown is clicked.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onMegaMenuClick(ev) {
+        const megaMenuToggleEl = ev.currentTarget;
+        // Ignore the event if the menus are hoverable and in desktop view (the
+        // hoverable menus are clicked on mobile view).
+        if (this.el.classList.contains("o_hoverable_dropdown")
+            && !megaMenuToggleEl.closest(".navbar.o_header_mobile")) {
+            return;
+        }
+        this._moveMegaMenu(megaMenuToggleEl);
+    },
+    /**
+     * Called when a mega menu dropdown is hovered.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onMegaMenuHover(ev) {
+        const megaMenuToggleEl = ev.currentTarget;
+        // Ignore the event if the menus are not hoverable or if we are in
+        // mobile view (again, the hoverable menus are clicked on mobile view).
+        if (!this.el.classList.contains("o_hoverable_dropdown")
+            || megaMenuToggleEl.closest(".navbar.o_header_mobile") && uiUtils.getSize() <= SIZES.SM) {
```

Comment:
It is needed because the hoverable dropdowns are considered "hoverable" above SM (check around in the file) so at MD they open on hover even though it is the `.o_header_mobile`

Note that it is something I find strange (to cut at SM when it is also the mobile menu at MD) , I already had it on my list of potential other bugs to fix about the header :wink: